File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/adl/format/media/StaticFormat.java
package com.kite9.k9server.adl.format.media;

/**
 * Used for CSS, XML, JS.
 *
 */
public class StaticFormat implements Format {
	
	private String extension;
	private MediaType[] mediaTypes;

	public StaticFormat(String extension, MediaType[] mediaTypes) {
		super();
		this.extension = extension;
		this.mediaTypes = mediaTypes;
	}
	
	@Override
	public MediaType[] getMediaTypes() {
		return mediaTypes;
	}

	@Override
	public String getExtension() {
		return extension;
	}

}

File: b/src/jvmMain/kotlin/com/kite9/pipeline/adl/format/media/StaticFormat.kt
package com.kite9.pipeline.adl.format.media

/**
 * Used for CSS, XML, JS.
 *
 */
data class StaticFormat(override val extension: String, override val mediaTypes: List<K9MediaType>) : Format {

    override val formatIdentifier: String
        get() =  mediaTypes[0].toString()

}